pkg/gui/controllers/helpers/merge_and_rebase_helper.go

```diff
@@ -224,8 +224,15 @@ func (self *MergeAndRebaseHelper) RebaseOntoRef(ref string) error {
 			Key:   's',
 			OnPress: func() error {
 				self.c.LogAction(self.c.Tr.Actions.RebaseBranch)
-				err := self.c.Git().Rebase.RebaseBranch(ref)
-				return self.CheckMergeOrRebase(err)
+				baseCommit := self.c.Modes().MarkedBaseCommit.GetSha()
+				err := lo.Ternary(baseCommit != "",
```

Comment:
I'm 99% sure this isn't working the way you expect: lo.Ternary evaluates both the if and else expressions before picking which one to return (just cos that's how the language works and lo has no way to get around that). So this will unconditionally call both rebase functions.

I believe lo has lo.TernaryFunc which you can use to only evaluate the expression if it's chosen, but at that point I wonder if it's just easier to use an if/else statement.

lo.Ternary should really only be used when it's dealing expressions that evaluate without side effects and without much time/memory footprint.

In this case I'd use an opts object in `RebaseBranch` and let it internally deal with the baseCommit arg.

Reply:
Ouch; fixed in c2afab33c5e3. I wonder why it worked though, I had used this locally for a while, and the tests are green too.

Reply:
Beats me. Maybe by some weird quirk of git logic they're mutually idempotent?

Reply:
Almost. What happened: when baseCommit was non-empty, it first did the rebase --onto, then the normal rebase afterwards. The latter succeeded too, but was a no-op, because the branch was already sitting on its target. Conversely, when baseCommit was empty, the first command would fail, and then the second command would run and do its job. Phew.

Reply:
Wow, crazy!